app/src/main/java/com/poupa/vinylmusicplayer/service/MusicService.java

```diff
@@ -463,13 +461,8 @@ public int getPosition() {
         return playingQueue.getCurrentPosition();
     }
 
-    public void playNextSong(boolean force) {
-        if (force && PreferenceUtil.getInstance().maintainSkippedSongsPlaylist()) {
```

Comment:
This is the main change: move the 'maintainSkippedSongs' processing to a lower-level utility method: playSongAt.

This allows sharing this processing with other sources of event (prev/gestures/...)